### PR TITLE
✨ Add track command to CLI

### PR DESCRIPTION
**Pull request by Koncopd**
`lndb track nb.ipynb --pypackage lamindb,bionty --parent parent_id`
The command initializes nbproject metadata in `nb.ipynb` to avoid writing it in `ln.track()`

It doesn't create `Transform` or `Run`, i think it is better to do from the notebook itself.

### CODE REVIEW

File: lndb/__main__.py [hunk 1]
```diff
@@ -87,6 +88,18 @@
...
 
+# track anotebook (init nbproject metadata)
+track_parser = subparsers.add_parser("track", help=track_help)
+aa = track_parser.add_argument
+nb_path_help = "A path to the notebook to track."
+aa("nb_path", type=str, metavar="filepath", help=nb_path_help)
+parent_help = (
+    "One or more (delimited by ',') ids of direct ancestors in a notebook pipeline"
+)
+aa("--parent", type=str, metavar="parent", default=None, help=parent_help)
```
Comment (falexwolf) [inline lndb/__main__.py:99]: I'd not support the parent argument as this is something that shouldn't be needed in the context of lamindb!

File: lndb/_notebook.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from typing import Optional
+
+from lamin_logger import logger
+
+
+def track(
+    notebook_path: str, parent: Optional[str] = None, pypackage: Optional[str] = None
```
Comment (falexwolf) [inline lndb/_notebook.py:7]: I'd remove the parent arg here!

File: lndb/_notebook.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+):
+    try:
```
Comment (falexwolf) [inline lndb/_notebook.py:9]: nbproject is a requirement of lamindb; hence, this isn't needed - but of course it doesn't hurt much! :)

File: lndb/_notebook.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        from nbproject.dev import initialize_metadata, read_notebook, write_notebook
+    except ModuleNotFoundError:
+        logger.warning("Install nbproject! pip install nbproject")
+        return None
+
+    nb = read_notebook(notebook_path)
+    if "nbproject" not in nb.metadata:
+        if parent is not None:
```
Comment (falexwolf) [inline lndb/_notebook.py:17]: This logic can disappear if we ditch parent here.

File: lndb/__main__.py [hunk 5]
```diff
@@ -87,6 +88,18 @@
 aa = migr.add_argument
 aa("action", choices=["generate"], help="Generate migration.")
 
+# track anotebook (init nbproject metadata)
+track_parser = subparsers.add_parser("track", help=track_help)
+aa = track_parser.add_argument
+nb_path_help = "A path to the notebook to track."
+aa("nb_path", type=str, metavar="filepath", help=nb_path_help)
```
Comment (falexwolf) [inline lndb/__main__.py:95]: Can we rename nb_path to filepath?

File: lndb/_notebook.py (hunk 2 above)
Comment (Koncopd) [reply lndb/_notebook.py]: Removed

File: lndb/_notebook.py (hunk 3 above)
Comment (Koncopd) [reply lndb/_notebook.py:7]: Yes, but not of lndb. Just to avoid any strange edge cases.

File: lndb/_notebook.py (hunk 4 above)
Comment (Koncopd) [reply lndb/_notebook.py]: Removed

File: lndb/__main__.py (hunk 5 above)
Comment (Koncopd) [reply lndb/__main__.py]: Renamed

**Pull request merged**